idaes/generic_models/properties/helmholtz/helmholtz.py

```diff
@@ -436,36 +438,112 @@ def basic_calculations(
         return blk, delta_liq, delta_vap, tau, x
 
     def s(self, **kwargs):
+        """Mixed phase entropy"""
         blk, delta_liq, delta_vap, tau, x = self.basic_calculations(**kwargs)
         self.add_funcs(names=["func_s"])
         s = blk.func_s(delta_liq, tau)*(1-x) + blk.func_s(delta_vap, tau)*x
         return s * self.param.uc_kJ_per_kgK_to_J_per_molK
 
+    def s_liq(self, **kwargs):
+        """Liquid phase entropy"""
+        blk, delta_liq, delta_vap, tau, x = self.basic_calculations(**kwargs)
+        self.add_funcs(names=["func_s"])
```

Comment:
Is there any issue with calling `add_funcs` multiple times with the same name? From what I can tell, `s`, `s_liq` and `s_Vap` would all call `self.add_funcs(names=["func_s"])`, so this code could be executed multiple times.

Reply:
It's ok.  That checks that the function doesn't exist before adding it.